<launch>

    <arg name="use_cable_sensors" default="true"/>
	<arg name="frame" default="data"/>
    <arg name="cable_unit_conversion"/>
    <arg name="cable_map_filename"/>
    <arg name="initial_estimate_filename"/>

	<node pkg="estimation" type="traj_estimation" name="traj_estimation" output="screen">
        <param name="red_endcaps_topic" value="/tensegrity/endcap/red/positions"/>
        <param name="blue_endcaps_topic" value="/tensegrity/endcap/blue/positions"/>
        <param name="green_endcaps_topic" value="/tensegrity/endcap/green/positions"/>
        <param name="tensegrity_pose_topic" value="/tensegrity/poses"/>
        <param name="tensegrity_traj_topic" value="/tensegrity/trajectory"/>
        <param name="sensors_topicname" value="/tensegrity/sensors"/>
        <!-- <param name="cables_topic" value="/tensegrity/cables"/> -->
        <param name="use_cable_sensors" value="$(arg use_cable_sensors)"/>
        <param name="frame" value="$(arg frame)"/>
        <param name="graph_resolution" value="1"/>
        <param name="optimization_frequency" value="10"/>
        <param name="publisher_frequency" value="5"/>
        <param name="trajectory_pub_frequency" value="2"/>
        <param name="cable_unit_conversion" value="$(arg cable_unit_conversion)"/>
        <param name="cable_map_filename" value="$(arg cable_map_filename)"/>
        <param name="initial_estimate_filename" value="$(arg initial_estimate_filename)"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="viz_traj_nodelet"  args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="BarsToMarkersStateEstimation" 
        args="load interface/TensegrityBarsToMarkers viz_traj_nodelet" output="screen">
        <param name="tensegrity_bars_topicname" value="/tensegrity/poses"/>
        <param name="tensegrity_markers_topicname" value="/tensegrity/markers"/>
        <param name="node_id" value="/nodes/BarsToMarkersStateEstimation"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="TensegrityTrajectoryToMarkers" 
        args="load interface/TensegrityTrajectoryToMarkers viz_traj_nodelet" output="screen">
        <param name="node_id" value="/nodes/TensegrityTrajectoryToMarkers"/>
        <param name="tensegrity_trajectory_topicname" value="/tensegrity/trajectory"/>
        <!-- <param name="tensegrity_markers_topicname" value="/tensegrity/markers"/> -->
        <param name="alpha" value="0.5"/>
    </node>
</launch>